### ISSUE TITLE
Does not work when Kubernetes has RBAC

### ISSUE THREAD

**Issue by kyomster** [estafette/estafette-cloudflare-dns#6]
Hello,

When trying to deploy on Kubernetes with RBAC I get:

{"time":"2017-11-08T22:03:56Z","severity":"error","app":"estafette-cloudflare-dns","version":"1.0.103","error":"kubernetes api: Failure 403 services is forbidden: User \"system:serviceaccount:default:default\" cannot list services at the cluster scope","message":"ListServices call failed"}

Is there a workaround ?

**Comment by etiennetremel** [comment]
Hi @kyomster, looks like the documentation is not up to date, I 'm going to update it later today. 

Until the fix come through, you have to manually create a service account:
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: estafette-cloudflare-dns
  namespace: estafette
```

Then add the `serviceAccount: estafette-cloudflare-dns` field to the deployment manifest, see:
https://github.com/estafette/estafette-cloudflare-dns/blob/master/kubernetes.yaml#L41

Let me know if you are still running into trouble/

**Comment by kyomster** [comment]
I apply the rbac.yaml
but I still get:

`[root@k8-master ~]# kubectl logs estafette-cloudflare-dns-6bd975b584-48q4b
{"time":"2017-11-12T14:49:15Z","severity":"info","app":"estafette-cloudflare-dns","version":"1.0.103","branch":"master","revision":"6f3b8458a1a60e57ccff8d65ab987bbed08cd26c","buildDate":"2017-09-12T15:44:38Z","goVersion":"go1.9","message":"Starting estafette-cloudflare-dns..."}
{"time":"2017-11-12T14:49:15Z","severity":"debug","app":"estafette-cloudflare-dns","version":"1.0.103","port":":9101","message":"Serving Prometheus metrics..."}
{"time":"2017-11-12T14:49:15Z","severity":"info","app":"estafette-cloudflare-dns","version":"1.0.103","message":"Listing services for all namespaces..."}
{"time":"2017-11-12T14:49:15Z","severity":"info","app":"estafette-cloudflare-dns","version":"1.0.103","message":"Watching services for all namespaces..."}
{"time":"2017-11-12T14:49:15Z","severity":"info","app":"estafette-cloudflare-dns","version":"1.0.103","message":"Watching ingresses for all namespaces..."}
{"time":"2017-11-12T14:49:15Z","severity":"error","app":"estafette-cloudflare-dns","version":"1.0.103","error":"kubernetes api: Failure 403 services is forbidden: User \"system:serviceaccount:default:estafette-cloudflare-dns\" cannot list services at the cluster scope","message":"ListServices call failed"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x8e1eb9]
goroutine 7 [running]:
main.main.func4(0xc420188270, 0xc42008a780, 0xc42001a030)
        /go/src/github.com/estafette/estafette-cloudflare-dns/main.go:225 +0xa39
created by main.main
        /go/src/github.com/estafette/estafette-cloudflare-dns/main.go:215 +0xa14
`

**Comment by etiennetremel** [comment]
Seems that the service account was created in the `default` namespace instead of `estafette`.

In case this still doesn't work, there is a PR with a chart that can ease the deployment of the app using Helm. You need to git clone the repo, checkout the branch `add-helm-chart` and follow the instructions provided in the _README.md_:
https://github.com/estafette/estafette-cloudflare-dns/pull/8

Let me know how that goes.

**Comment by kyomster** [comment]
It's working now, thanks.
But just realized that it's working only for Services and Ingresses, I thought it would work with Pods.

**Comment by etiennetremel** [comment]
Services are mandatory to serve external traffic to a pod.
This controller only generate certificate via LetsEncrypt and store it into a secret. In your pod you can setup a sidecar container with Nginx and configure it to consume the secret.

**Comment by JorritSalverda** [comment]
The released Helm chart `1.2.0` has the service account in there by default. See readme or https://helm.estafette.io for instructions. Resolving this in milestone `1.2.1` for bookkeeping.
